internal/services/workloads/workloads_sap_virtual_instance_resource.go

```diff
@@ -0,0 +1,1122 @@
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"central_server_virtual_machine_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: computeValidate.VirtualMachineID,
+					},
+
+					"managed_storage_account_name": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ForceNew:     true,
+						ValidateFunc: storageValidate.StorageAccountName,
+					},
+				},
+			},
+		},
+
+		"managed_resource_group_name": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: resourcegroups.ValidateName,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r WorkloadsSAPVirtualInstanceResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r WorkloadsSAPVirtualInstanceResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 60 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model WorkloadsSAPVirtualInstanceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.Workloads.SAPVirtualInstances
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := sapvirtualinstances.NewSapVirtualInstanceID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			identity, err := identity.ExpandUserAssignedMapFromModel(model.Identity)
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+
+			parameters := &sapvirtualinstances.SAPVirtualInstance{
+				Identity: identity,
+				Location: location.Normalize(model.Location),
+				Properties: sapvirtualinstances.SAPVirtualInstanceProperties{
+					Environment: sapvirtualinstances.SAPEnvironmentType(model.Environment),
+					SapProduct:  sapvirtualinstances.SAPProductType(model.SapProduct),
+				},
+				Tags: &model.Tags,
+			}
+
+			if v := model.DeploymentWithOSConfiguration; v != nil {
+				deploymentWithOSConfiguration, err := expandDeploymentWithOSConfiguration(v)
+				if err != nil {
+					return err
+				}
+				parameters.Properties.Configuration = deploymentWithOSConfiguration
+			}
+
+			if v := model.DiscoveryConfiguration; v != nil {
+				parameters.Properties.Configuration = expandDiscoveryConfiguration(v)
+			}
+
+			if v := model.ManagedResourceGroupName; v != "" {
+				parameters.Properties.ManagedResourceGroupConfiguration = &sapvirtualinstances.ManagedRGConfiguration{
+					Name: utils.String(v),
+				}
+			}
+
+			if _, err := client.Create(ctx, id, *parameters); err != nil {
```

Comment:
could we use `CreateThenPoll` here?

Reply:
Nope. Because we need to [check provisioning state](https://github.com/hashicorp/terraform-provider-azurerm/pull/22869/files#diff-52bff945012c533329f22b42f0ed771fe943f3c8f0847b9a44d1c8068a7af560R690) in LRO otherwise it would fail.

Reply:
The [PR](https://github.com/hashicorp/go-azure-sdk/pull/704) is submitted for this.

Reply:
Updated to use createThenPoll